pulp-glue/pulp_glue/common/context.py

```diff
@@ -153,13 +156,29 @@ def _patch_api_spec(self) -> None:
                 patched_python_remote_serializer["properties"][prop]["type"] = "array"
                 patched_python_remote_serializer["properties"][prop]["items"] = {"type": "string"}
 
+    @property
+    def domain_enabled(self) -> bool:
+        return self.api.api_spec.get("info", {}).get("x-pulp-domain-enabled", False)  # type: ignore
+
+    @property
+    def pulp_domain(self) -> str:
+        return self._domain or "default"
+
+    @property
+    def api_path(self) -> str:
+        if self.domain_enabled:
+            return self._api_root + self.pulp_domain + "/api/v3/"
+        return self._normal_api_path
+
     @property
     def api(self) -> OpenAPI:
         if self._api is None:
             if self._api_kwargs.get("username") and not self._api_kwargs.get("password"):
                 self._api_kwargs["password"] = self.prompt("password", hide_input=True)
             try:
-                self._api = OpenAPI(doc_path=f"{self.api_path}docs/api.json", **self._api_kwargs)
+                self._api = OpenAPI(
+                    doc_path=f"{self._normal_api_path}docs/api.json", **self._api_kwargs
```

Comment:
Yep, api-docs will also be available in each domain.

Reply:
But still we need to assemble the string manually, because the `api_root` property would call to the server giving us an infinite loop. Also when using the non-domained api docs, we only save a local copy once (pretty much preferred).